modules/tinymce/src/core/main/ts/caret/ClosestCaretCandidate.ts

```diff
@@ -51,22 +51,42 @@ const clientInfo = (rect: NodeClientRect, clientX: number): FakeCaretInfo => {
   };
 };
 
-const horizontalDistance: DistanceFn = (rect, x, _y) => Math.min(Math.abs(rect.left - x), Math.abs(rect.right - x));
+const horizontalDistance: DistanceFn = (rect, x, _y) => x > rect.left && x < rect.right ? 0 : Math.min(Math.abs(rect.left - x), Math.abs(rect.right - x));
```

Comment:
Maybe add a comment here so explain this? Also the line is pretty long now so we may want to split it over 2 lines. Something like this maybe?
```suggestion
// Measure the distance between the x and the closest edge of the rect.
// If the x is inside the rect then always return 0.
const horizontalDistance: DistanceFn = (rect, x, _y) =>
  x > rect.left && x < rect.right ? 0 : Math.min(Math.abs(rect.left - x), Math.abs(rect.right - x));
```